### ISSUE TITLE
Recipe to migrate the use of Ehcache 2.0 to 3.0 in Hibernate

### ISSUE THREAD

**Issue by tkvangorder** [openrewrite/rewrite-migrate-java#162]
Hibernate has deprecated the use of the `ehcache` 2.0 in favor of using `jcache` (with the `ehcache` 3.0 as the underlying implementation)

This recipe should:

- Replace `hibernate-ehcache` dependency with `hibernate-jcache`
- It should replace the `hibernate.cache.region.factory_class` property with `jcache`
- It should add the following two properties : 

```
<property name="hibernate.javax.cache.provider" value="org.ehcache.jsr107.EhcacheCachingProvider"/>
<property name="hibernate.javax.cache.uri" value="file:/META-INF/ehcache.xml"/>
```

And finally if the `ehcache.xml` file does not exist, it should create that file with a defaultCache entry: 

```
<ehcache>
    <defaultCache maxEntriesLocalHeap="1000" eternal="false" timeToIdleSeconds="300" timeToLiveSeconds="600">
    </defaultCache>
</ehcache>
```

**Comment by knutwannheden** [comment]
This is for Hibernate 6.0, right?

**Comment by tkvangorder** [comment]
Yes, this is for the move to Hibernate 6.0. I believe this approach could be applied to versions of Hibernate 5.4 and higher. I honestly have not worked with Hibernate in a very long time, but I ran across this as part of the work in #157 (The `hibernate-ehcache` no longer exists in Hibernate 6.)
